### ISSUE TITLE
Warns about no value for an input, although a value has been set

### ISSUE THREAD

**Issue by multani** [inspec/inspec#4769]
# Describe the problem

I created a test profile with this `inspec.yml` configuration:
```yaml
name: default
inputs:
  - name: foo
    type: numeric
    required: true

  - name: bar
    type: string
    required: true
```

I have a simple control which get the values of these (mandatory) inputs and asserts some values based on these inputs:
```ruby
foo = input('foo')
bar = input('bar')
...
# omitted but foo and bar are used
```

I have a input variable file with:

```yaml
foo: 42
bar: hello
```

When I ran InSpec, I get warnings about missing values for the inputs (+ the confusing message from #4445):

```
$ bundle exec inspec exec profile --input-file $PWD/test-inputs.yml 
[2019-12-06T14:17:14+01:00] WARN: Input 'foo' does not have a value. Use --input-file to provide a value for 'foo' or specify a  value with `attribute('foo', value: 'somevalue', ...)`.
[2019-12-06T14:17:14+01:00] WARN: Input 'bar' does not have a value. Use --input-file to provide a value for 'bar' or specify a  value with `attribute('bar', value: 'somevalue', ...)`.
...
```

See https://github.com/multani/inspec-issue-4769 for a demonstration of the problem.

**Comment by chiefy** [comment]
```
inspec exec --self-signed /var/opt/inspec/profiles/local/inspec-profile-eks -t aws:// --input awsregion=us-east-1 clustername=xyz mineksversion=13
[2020-04-10T17:14:12+00:00] WARN: Input 'mineksversion' does not have a value. Use --input-file to provide a value for 'mineksversion' or specify a  value with `attribute('mineksversion', value: 'somevalue', ...)`.
```

**Comment by james-stocks** [comment]
This is a pretty common pattern - a profile declares an input that must be user-specified. There is no valid default value that can be given in `inspec.yml` e.g. an API key

**Comment by clintoncwolfe** [comment]
I think this may be more subtle. I just tried it in the `undeclared` test fixture and could not reproduce the issue, and learned that an essential requirement to trigger the bug is to declare the inputs in the metadata file - simply implicitly referencing them in the control file is not enough.

**Comment by james-stocks** [comment]
Fixed in 4.18.111
